com.unity.shadergraph/Editor/Drawing/Views/PropertyNodeView.cs

```diff
@@ -1,16 +1,32 @@
 ﻿using System;
 using System.Linq;
+using System.Reflection;
+using Drawing.Inspector;
 using UnityEditor.Experimental.GraphView;
 using UnityEditor.Graphing;
+using UnityEditor.Graphing.Util;
 using UnityEditor.Rendering;
 using UnityEditor.ShaderGraph.Drawing;
+using UnityEditor.ShaderGraph.Drawing.Controls;
+using UnityEditor.ShaderGraph.Internal;
 using UnityEngine;
 using UnityEngine.UIElements;
+using Data.Interfaces;
+using Drawing.Inspector.PropertyDrawers;
 
 namespace UnityEditor.ShaderGraph
 {
-    sealed class PropertyNodeView : TokenNode, IShaderNodeView
+    sealed class PropertyNodeView : TokenNode, IShaderNodeView, IInspectable
     {
+        // When the properties are changed, this delegate is used to trigger an update in the view that represents those properties
+        private Action m_propertyViewUpdateTrigger;
+        private ShaderInputPropertyDrawer.ChangeReferenceNameCallback m_resetReferenceNameTrigger;
+
+        static Type s_ContextualMenuManipulator = AppDomain.CurrentDomain.GetAssemblies().SelectMany(x => x.GetTypesOrNothing()).FirstOrDefault(t => t.FullName == "UnityEngine.UIElements.ContextualMenuManipulator");
```

Comment:
Can this use `TypeCache` instead? Maybe not but always prefer the cache over iterating the Assemblies since that's really slow. At least it's static so it only happens once.

Reply:
I can use the TypeCache to get the types derived from the parent class of the ContextualMenuManipulator and then do the same name check to find it, code looks like this now:

`TypeCache.GetTypesDerivedFrom<MouseManipulator>().FirstOrDefault(t => t.FullName == "UnityEngine.UIElements.ContextualMenuManipulator");`